lib/KFocusTrap.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+
+  <div>
+    <div
+      v-if="!disabled"
+      class="focus-trap-first"
+      tabindex="0"
+      @focus="handleFirstTrapFocus"
+    ></div>
+
+    <slot></slot>
+
+    <div
+      v-if="!disabled"
+      class="focus-trap-last"
+      tabindex="0"
+      @focus="handleLastTrapFocus"
+    ></div>
+  </div>
+
+</template>
+
+
+<script>
+
+  /**
+   * This component ensures that focus moves between the first element
+   * and the last element of content provided by the default slot.
+   * In disabled state, it only renders whatever has been passed
+   * to the default slot, and doesn't add any focus trap behavior,
+   * allowing for flexible use from parent components.
+   */
+  export default {
+    name: 'FocusTrap',
+    props: {
+      disabled: {
```

Comment:
Would you document this prop? You can see other components.

The docstring will then show in the auto-generated `props` documentation on a documentation page for this component.